src/directives/show.js

```diff
@@ -1,55 +1,58 @@
-import { transitionIn, transitionOut } from '../utils'
+import { transitionIn, transitionOut } from '../transitions'
+import { showElement, hideElement } from '../utils'
 
 export function handleShowDirective(component, el, value, modifiers, initialUpdate = false) {
-    const hide = () => {
-        el.style.display = 'none'
+    // Resolve any previous pending transitions before starting a new one
+    if (el.__x_transition_remaining && el.__x_transition_last_value !== value) {
+        el.__x_transition_remaining()
     }
 
-    const show = () => {
-        if (el.style.length === 1 && el.style.display === 'none') {
-            el.removeAttribute('style')
-        } else {
-            el.style.removeProperty('display')
-        }
-    }
-
-    if (initialUpdate === true) {
+    // Resolve immediately if initial page load
+    if (initialUpdate) {
         if (value) {
-            show()
+            showElement(el)
         } else {
-            hide()
+            hideElement(el)
         }
         return
     }
 
     const handle = (resolve) => {
         if (! value) {
             if ( el.style.display !== 'none' ) {
-                transitionOut(el, () => {
-                    resolve(() => {
-                        hide()
-                    })
-                }, component)
+                transitionOut(el, component, () => {
+                    // If previous transitions still there, don't use resolve
+                    if (el.__x_transition_remaining) {
```

Comment:
Similar to the other PR, I'm sure we need it but I can't understand why. Can you explain please?

Reply:
also the test passes master, I think we need a better one.

Reply:
- Oh It's the where *black hole* begins. In case we have a transition going on but if value did not change since last time, we should not resolve it in that case. This part is related to `immediate` and `hide child elements before parent` related thing. I got lost a lot here. Even while i am explaining to you now :)
- Test has been updated and now master fails these but PRs passes. There was a timing issue where master flicks :) So i needed to find out best time the falsy thing happens.